lib/Sema/CodeSynthesisDistributedActor.cpp

```diff
@@ -763,6 +762,30 @@ addDistributedActorCodableConformance(
     return nullptr;
   }
 
+  if (actor->isGeneric()) {
+    auto idTy = C.getAssociatedTypeOfDistributedSystemOfActor(actor, C.Id_ActorID);
+    if (idTy->hasError()) {
+      return nullptr;
+    }
+    auto encodableConf = module->lookupConformance(
+        idTy, C.getProtocol(swift::KnownProtocolKind::Encodable),
+        /*allowMissing=*/true);
+    auto decodableConf = module->lookupConformance(
+        idTy, C.getProtocol(swift::KnownProtocolKind::Decodable),
+        /*allowMissing=*/true);
+
+    // the system's ID is not codable, thus the actor isn't as well -- don't add the conformance.
+    if (encodableConf.isInvalid()) {
+      return nullptr;
+    }
+    if (decodableConf.isInvalid()) {
+      return nullptr;
+    }
+  }
+
+  assert(proto->isSpecificProtocol(swift::KnownProtocolKind::Decodable) ||
+         proto->isSpecificProtocol(swift::KnownProtocolKind::Encodable));
+
   // === Does the actor explicitly conform to the protocol already?
   auto explicitConformance =
       module->lookupConformance(actor->getInterfaceType(), proto);
```

Comment:
This will always return false because getInterfaceType() returns a metatype given a TypeDecl